initial commit + done HW1

drag0nfet opened the pull request:
Added 2 animals and User.
Modified Solution for testing User

drag0nfet committed:
initial commit + done HW
Added 2 animals and User.
Modified Solution for testing User

drag0nfet committed:
Fix Solution and expansion of Flea
- Solution now has correct data for User
- Flea can heal now

lefasdayan commented on ru/turing/courses/lesson2/Solution.java:
```diff
@@ -1,4 +1,41 @@
+    public static void main(String[] args) {
+        // Task-1
+        String name1 = "Aboba";
+        User aboba = getUser(name1);
```
А зачем заводить отдельный метод с отдельными переменными, который возвращает пользователя, если можно просто объявить пользователя через конструктор?

lefasdayan commented on ru/turing/courses/lesson2/Solution.java:
```diff
@@ -1,4 +1,41 @@
+        User aboba = getUser(name1);
+        ZoneId z = ZoneId.of( "Europe/Moscow" );
+        ZonedDateTime date2 = ZonedDateTime.now(z);
+        String date = DateTimeFormatter.ofPattern("dd/MM/yyyy").format(date2);
```
Все эти переменные есть у тебя в методах, здесь они не нужны

lefasdayan commented on ru/turing/courses/lesson2/Solution.java:
```diff
@@ -1,4 +1,41 @@
+
+        // Task-2
+
+        // реализации классов (abstract)Animal, Flea, Donkey
```
Было бы прикольно увидеть реализацию этих классов здесь же

lefasdayan commented on ru/turing/courses/lesson2/Solution.java:
```diff
@@ -1,4 +1,41 @@
+        // реализации классов (abstract)Animal, Flea, Donkey
+    }
+
+    private static User getUser(String name1) {
```
Убрать и заменить на конструктор

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+    protected String dateOfBirth;
+    protected int dayOfBirth;
+    protected int monthOfBirth;
+    protected int yearOfBirth;
```
День, месяц и год не нужны как поля, если ты вытаскиваешь их из строки dateOfBirth.

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+        int y = Integer.parseInt(date.substring(6, 10)) - yearOfBirth, m = Integer.parseInt(date.substring(3, 5));
+        if (m <= monthOfBirth) {
+            if (m < monthOfBirth || Integer.parseInt(date.substring(0, 1)) < dayOfBirth)
+                y--;
```
Не работает :)
Попробуй сделать через LocalDate и его методы по вычислению лет/месяцев между двумя датами

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+    protected String flat;
+
+    protected void printYearsLived() {
+        ZoneId z = ZoneId.of( "Europe/Moscow" );
```
Чтобы пробелы были только там, где нужно, можно использовать Ctrl-Alt-L

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+
+    protected void printYearsLived() {
+        ZoneId z = ZoneId.of( "Europe/Moscow" );
+        ZonedDateTime date2 = ZonedDateTime.now(z);
```
Переназови переменные, чтобы названия несли какой-то смысл, потому что date1, date2, z, date, fooo и подобные не помогают понять, что происходит

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+
+    public void printAdress() {
+        String ans;
+        ans = name + " lives in " + house + " " + street + " str., " + flat +
```
Можно объединить объявление и присвоение: String ans = ...
А еще нет перевода на новую строку, но ладно)

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+    public User() {
+    }
+
+    public void setDateOfBirth(String dateOfBirth) {
```
Ты в итоге так и не использовал этот метод, тогда его можно не оставлять

lefasdayan commented on ru/turing/courses/lesson2/User.java:
```diff
@@ -0,0 +1,161 @@
+        this.flat = flat;
+    }
+
+    public User() {
```
Если не используешь какой-нибудь конструктор, то можно его не оставлять

drag0nfet committed:
2nd fix, mainly for Solution, secondly for another modules
- Исправил 1-е задание, была глупая ошибка, брал 2 раза текущую дату
- Теперь текущая дата - статический метод класса User
- Мертвые животные разучились разговаривать
- Бессмертной блохе можно сломать колени
- Добавлена шуточная реализация для ослов и блох в Solution

lefasdayan commented on ru/turing/courses/lesson2/Solution.java:
```diff
@@ -1,4 +1,85 @@
+            justJump.tryJump(3 / (Math.random() * 1.5));
+        }
+    }
+    public static void fleaFight(Flea character, Flea main) {
```
Если метод относится к блохам, то лучше в класс с блохами его и добавить)

drag0nfet committed:
3rd final fix, moved the own func to the class it belongs to
- Битва с блохой теперь реализована как собственный метод класса блохи

lefasdayan commented:
Молодец!!!
